DOCUMENTATION.md

```diff
@@ -1370,529 +1395,202 @@ The following are example configurations.
+Most of the functions in the API return `Promise`. Therefore, you need to be familiar with the concept of promises to use the Node.js client. If not, you can learn about them using various online resources, e.g., the [Promise JS](https://www.promisejs.org/) website.
 
-#### 6.1.2. TLS/SSL for Hazelcast Node.js Clients
+Promises provide a better way of working with callbacks. You can chain asynchronous functions by the `then()` function of promise. Also, you can use `async/await`, if you use Node.js 8 and higher versions.
 
-TLS/SSL for the Hazelcast Node.js client can be configured using the `SSLConfig` class. In order to turn it on, `enabled` property of `SSLConfig` should be set to `true`:
+If you are ready to go, let's start to use Hazelcast Node.js client.
 
-```javascript
-var fs = require('fs');
+The first step is the configuration. You can configure the Node.js client declaratively or programmatically. We will use the programmatic approach throughout this chapter. See the [Programmatic Configuration section](#311-programmatic-configuration) for details.
+
+The following is an example on how to create a `ClientConfig` object and configure it programmatically:
 
+```javascript
 var clientConfig = new Config.ClientConfig();
-var sslConfig = new Config.SSLConfig();
-sslConfig.enabled = true;
-clientConfig.networkConfig.sslConfig = sslConfig;
+clientConfig.clusterName = 'dev';
+clientConfig.networkConfig.addresses.push('10.90.0.1', '10.90.0.2:5702');
 ```
 
-`SSLConfig` object takes various SSL options defined in the [Node.js TLS Documentation](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback). You can set your custom options
-object to `sslConfig.sslOptions`.
-
-#### 6.1.3. Mutual Authentication
+The second step is initializing the `HazelcastClient` to be connected to the cluster:
 
-As explained above, Hazelcast members have key stores used to identify themselves (to other members) and Hazelcast clients have trust stores used to define which members they can trust.
+```javascript
+Client.newHazelcastClient(clientConfig).then(function (client) {
+    // some operation
+});
+```
 
-Using mutual authentication, the clients also have their key stores and members have their trust stores so that the members can know which clients they can trust.
+**This client object is your gateway to access all the Hazelcast distributed objects.**
 
-To enable mutual authentication, firstly, you need to set the following property on the server side in the `hazelcast.xml` file:
+Let's create a map and populate it with some data, as shown below.
 
-```xml
-<network>
-    <ssl enabled="true">
-        <properties>
-            <property name="javax.net.ssl.mutualAuthentication">REQUIRED</property>
-        </properties>
-    </ssl>
-</network>
+```javascript
+var map;
+// Get the Distributed Map from Cluster.
+client.getMap('my-distributed-map').then(function (mp) {
+    map = mp;
+    // Standard Put and Get.
```

Comment:
nit: some of such comments end with dot, some not. Some of them start with Upper case, some not. Ideally we should make them consistent.